packages/mobile/src/screens/app-screen/AppTabScreen.tsx

```diff
@@ -155,14 +160,27 @@ export const AppTabScreen = ({ baseScreen, Stack }: AppTabScreenProps) => {
     [drawerNavigation]
   )
 
+  /**
+   * Reset lastNavAction on transitionEnd
+   * Need to do this via screenListeners on the Navigator because listening
+   * via navigation.addListener inside a screen does not always
+   * catch events from other screens
+   */
+  const handleTransitionEnd = useCallback(() => {
+    lastNavAction = undefined
+  }, [])
+
   useEffect(() => {
     drawerNavigation?.setOptions({ swipeEnabled: !isNowPlayingDrawerOpen })
   }, [drawerNavigation, isNowPlayingDrawerOpen])
 
   return (
     <Stack.Navigator
       screenOptions={screenOptions}
-      screenListeners={{ state: handleChangeState }}
+      screenListeners={{
```

Comment:
would memozing this object help at all for general perf you think?

Reply:
Hard to say, depends on how often this gets rendered. We could dig in and see but I would say it's ok to leave as is for now